host/volume/manager/manager.go

```diff
@@ -279,12 +296,12 @@ func (m *Manager) getProviderBucket(tx *bolt.Tx, providerID string) (*bolt.Bucke
 	return providerBucket, err
 }
 
-// Called to sync changes to disk when a VolumeInfo is updated
-func (m *Manager) persistVolume(tx *bolt.Tx, id string) error {
+// Called to sync changes to disk when a volume is updated
+func (m *Manager) persistVolume(tx *bolt.Tx, vol volume.Volume) error {
 	// Save the general volume info
 	volumesBucket := tx.Bucket([]byte("volumes"))
-	k := []byte(id)
-	vol, ok := m.volumes[id]
+	k := []byte(vol.Info().ID)
+	_, ok := m.volumes[vol.Info().ID]
 	if !ok {
```

Comment:
It's used again further down in the branch for whether or not the provider storage is doing a deletion.  I'll rename so we don't have an 'ok' floating around that long.